### DISCUSSION TITLE
sop/quickStart

### DISCUSSION THREAD

**Discussion by giscus** [ATQQ/sugar-blog#61] [Announcements]
# sop/quickStart

1分钟内完成自己的博客创建

https://theme.sugarat.top/sop/quickStart.html

**Comment by lucky-cmyk** [comment]
SyntaxError: Invalid regular expression: /^docs\/: \ at end of pattern

**Reply by lucky-cmyk** [reply]
win 走到 dev 这步骤报错

**Reply by ATQQ** [reply]
我晚上找个win机器看一下，感谢反馈

**Reply by ATQQ** [reply]
这个bug修了
可以重新拉模板，或者更新一下现有模板里的依赖

```sh
pnpm add @sugarat/theme@latest
```

**Comment by rainlam163** [comment]
6，学习了

**Comment by ttmars** [comment]
666

**Comment by sword-demon** [comment]
win7 下执行 `pnpm run build` 提示内存不够

```bash
memory allocation of 62446620 bytes failed
```

不知道是这个电脑垃圾还是什么，我换成我的`mac`就屁事没有

**Reply by ATQQ** [reply]
你Windows电脑 几个G内存呀。可以设置node构建占用的内存上限。

安装的时候有个默认值 可能设置小了

**Reply by sword-demon** [reply]
win7的是 8G 内存，今天试了 `node --max_old_space_size=10240  vitepress build docs` 也没啥用

**Reply by ATQQ** [reply]
> win7的是 8G 内存，今天试了 `node --max_old_space_size=10240  vitepress build docs` 也没啥用

那就是 配得不对吧 没生效这样配 试试通过环境变量配置

GPT4 的建议

在Windows平台上运行Node.js并设置内存占用上限的步骤与在其他操作系统上类似。以下是具体步骤：

打开命令提示符（CMD）或PowerShell。
如果你想要查看Node.js的V8引擎默认的内存限制，可以运行以下命令：

 
node --v8-options | findstr "max_old_space_size"
 

在Windows平台上，您可以通过设置环境变量来改变Node.js进程的内存限制。这可以通过命令提示符（CMD）或PowerShell完成。以下是如何设置环境变量的步骤：

打开命令提示符或PowerShell。
使用以下命令设置环境变量，以改变Node.js的最大老生代（old space）内存大小。例如，如果您想要设置内存上限为4GB，可以使用以下命令：

在命令提示符（CMD）中：

 
set NODE_OPTIONS=--max-old-space-size=4096
 

在PowerShell中：

 
$env:NODE_OPTIONS="--max-old-space-size=4096"

**Reply by sword-demon** [reply]
`set NODE_OPTIONS=--max-old-space-size=4096`设置完这个，我直接干啥所有的都是
`<--- Last few GCs --> <--JS Stacktrace --> FATAL JAVASCRIPT OOM in GC during deserialization`

**Reply by ATQQ** [reply]
这。。原来是多大 ni设置回去

**Reply by ATQQ** [reply]
可能是兼容性问题，你 Win7 上 node.js 版本多少。看描述 >=14 都要 Win10了

```sh
node -v
```

**Reply by ATQQ** [reply]
Node.js 10.x: 支持 Windows 7, Windows Server 2008 R2, 或更高版本。
Node.js 12.x: 支持 Windows 7, Windows Server 2008 R2, 或更高版本。
Node.js 14.x: 支持 Windows 8.1, Windows Server 2012 R2, 或更高版本。
Node.js 15.x 和以上: 支持 Windows 8.1, Windows Server 2012 R2, 或更高版本。

**Comment by sword-demon** [comment]
大佬，为啥我`pnpm run build`之后部署到服务器之后，侧边栏都是404，就是没有自动带上`.html`后缀，其他地址都对的

**Reply by ATQQ** [reply]
是更新到最新的版本了吗主题？

不要设置cleansUrl

**Comment by yxb-tech** [comment]
可以生成js文件类型的项目吗

**Reply by ATQQ** [reply]
可以，我记一下，回头加一加。

我看现在的模板里没什么特殊的TS语法，你可以吧 ts,mts 换成 js mjs 即可应该

**Reply by yxb-tech** [reply]
十分感谢！我把 ts,mts 换成 js mjs，项目成功启动。
我是后端开发，前端知识了解较少，且idea无法识别mts文件语法。。。

**Reply by ATQQ** [reply]
好滴好滴 👍🏻，我记一下这个 回头加一下纯js的模板 ，到时候滴滴滴你

**Comment by andyxiong11** [comment]
大佬，项目里面的内置插件页面跳转链接提示“oml2d.com”域名过期了

**Reply by ATQQ** [reply]
这就改一下：https://oml2d.hacxy.cn/

**Comment by Artemik-web** [comment]
111
